### PR TITLE
Now all C tests can handle extra tasks.

### LINKED ISSUE #211: C tests can't really handle more than 4 tasks...
The C tests are supposed to be written so that more than the specified number of tasks can be used.

But in fact, when I run a 4-task test with 6 processors, it hangs at the end of the test and does not return.

### PR DESCRIPTION
**Pull request by edhartnett**
Fixes #211.

There are only test code changes in this PR - no library code changes.

Many tests could already handle extra tasks, but some could not, because they were calling InitIntracomm with MPI_COMM_WORLD (instead of test_comm), and/or using ntasks for calculations, instead of TARGET_NTASKS.

Those don't matter if they run on exactly the correct number of tasks, but they cause failures if extra tasks are included.

To test all C tests are now run with an extra task.

Also removes doxygen tags from test code (since we don't run doxygen on tests).

### CODE REVIEW

File: tests/unit/test_intercomm2.c [hunk 1]
```diff
@@ -212,8 +198,7 @@ check_file(int iosysid, int format, char *filename, int my_rank, int verbose)
         ERR(ERR_WRONG);
     if ((ret = PIOc_get_att_int(ncid, NC_GLOBAL, ATT_NAME, &att_data)))
         ERR(ret);
-    if (verbose)
-        printf("%d test_intercomm2 att_data = %d\n", my_rank, att_data);
+    printf("%d test_intercomm2 att_data = %d\n", my_rank, att_data);
```
Comment (jedwards4b) [inline tests/unit/test_intercomm2.c]: why did you remove the verbose flag?

Comment (edhartnett) [inline tests/unit/test_intercomm2.c]: Because the build system already provides the feature that tests will run without output. I expected originally that my test print statements would pollute the output with masses of lines of information, so I would have to turn them off with a verbose flag.

But since I don't need to do that, it was always set to 1 and was therefore just useless code.
